File: app/src/main/java/com/fyp/agrifarm/app/news/db/Converter.java
package com.fyp.agrifarm.app.news.db;

import androidx.room.TypeConverter;

import java.util.ArrayList;
import java.util.Arrays;


public class Converter {


    @TypeConverter
    public static ArrayList<String> fromStringToArray(String value) {
        return new ArrayList<>(Arrays.asList(value.split("#")));
    }

    @TypeConverter
    public static String arrayToString(ArrayList<String> arrayList) {
        StringBuilder builder = new StringBuilder();
        for (String item : arrayList) {

            builder.append(item);
            builder.append("#");
        }
        builder.deleteCharAt(builder.lastIndexOf("#"));
        return builder.toString();
    }

}

File: app/src/main/java/com/fyp/agrifarm/app/news/db/Converter.kt
package com.fyp.agrifarm.app.news.db

import androidx.room.TypeConverter
import java.util.*

object Converter {
    @JvmStatic
    @TypeConverter
    fun fromStringToArray(value: String): List<String> {
        return listOf(*value.split("#@#".toRegex()).toTypedArray())
    }

    @JvmStatic
    @TypeConverter
    fun arrayToString(list: List<String?>): String {
        val builder = StringBuilder()
        for (item in list) {
            builder.append(item)
            builder.append("#@#")
        }
        builder.deleteCharAt(builder.lastIndexOf("#@#"))
        return builder.toString()
    }
}